operator/internal/status/status.go

```diff
@@ -59,6 +59,31 @@ func Refresh(ctx context.Context, k k8s.Client, req ctrl.Request, now time.Time)
 		} else {
 			stack.Status.Conditions[index] = condition
 		}
+
+		for _, sc := range stack.Status.Storage.Schemas {
+			if sc.SchemaStatus == "" {
+				continue
+			}
+			var reason lokiv1.LokiStackConditionReason
+			switch sc.SchemaStatus {
+			case StorageSchemaOutOfRetention:
+				reason = lokiv1.ReasonSchemaOutOfRetention
+			case StorageSchemaNeedsUpgrade:
+				reason = lokiv1.ReasonSchemaUpgradeRecommended
+			}
+			warning := metav1.Condition{
+				Type:               string(lokiv1.ConditionWarning),
+				Message:            sc.SchemaStatus,
+				Reason:             string(reason),
+				Status:             metav1.ConditionTrue,
+				LastTransitionTime: metav1.Now(),
+			}
+
+			if warningExists(warning, stack.Status.Conditions) {
+				continue
+			}
+			stack.Status.Conditions = append(stack.Status.Conditions, warning)
+		}
```

Comment:
This does not work as expected. The warning is added with `status: true` but on subsequent updated the status is reset to false, rendering the condition inactive. I think re-using the "support multiple conditions" [commit](https://github.com/grafana/loki/commit/f390f3734f5a123affe5f6438888b7f9a5ab6387) from my spike might be a good candidate.

Reply:
Yeah thanks, I forgot to change this one